compiler/hash-typecheck/src/error.rs

```diff
@@ -67,6 +67,14 @@ pub enum TcError {
         args: Args,
         unification_errors: Vec<TcError>,
     },
+    InvalidElementOfMerge {
+        term: TermId,
+    },
+    MergeShouldOnlyContainOneNominal {
+        merge_term: TermId,
+        nominal_term: TermId,
+        second_nominal_term: TermId,
+    },
```

Comment:
I wonder how these will become human-readable messages :)

Reply:
If the concept of a merge is known by the user, it shouldn't be difficult to generate understandable error messages for this.